packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -545,15 +545,26 @@ function makeswiftApiClientSyncMiddleware(
 export type Store = ReduxStore<State, Action> & { dispatch: Dispatch }
 
 export function configureStore({
+  rootElements,
   preloadedState,
   client,
 }: {
+  rootElements?: Map<string, Documents.Element>
   preloadedState?: PreloadedState<State>
   client: MakeswiftClient
 }): Store {
+  const initialState = {
+    ...preloadedState,
+    documents: Documents.getInitialState({ rootElements }),
+  }
+
+  if (typeof window === 'undefined') {
```

Comment:
I feel like this logic should be colocated with the browser API dependencies. It feels weird to magically put this check here since there's nothing about the other middleware that indicate they require browser APIs. I think the better thing to do would be to check for this right before a browser-only API is used

Reply:
So the reason why I put it here is that even though they might not use browser-only API, all other middlewares are only needed in the browser; they are not needed for the SSR. 

I was hoping that we could reduce the amount of code needed for the SSR/page generation even by a small amount.